browser-switch/src/main/java/com/braintreepayments/api/BrowserSwitchClient.java

```diff
@@ -76,7 +78,46 @@ public void start(@NonNull FragmentActivity activity, @NonNull BrowserSwitchOpti
         }
     }
 
-    void assertCanPerformBrowserSwitch(FragmentActivity activity, BrowserSwitchOptions browserSwitchOptions) throws BrowserSwitchException {
+    /**
+     * Open a browser or <a href="https://developer.chrome.com/multidevice/android/customtabs">Chrome Custom Tab</a>
+     * with a given set of {@link BrowserSwitchOptions} from an Android activity.
+     *
+     * @param activity the activity used to start browser switch
+     * @param browserSwitchOptions {@link BrowserSwitchOptions} the options used to configure the browser switch
+     * @return a pending {@link BrowserSwitchRequest} that should be stored and passed to
+     * {@link BrowserSwitchClient#parseResult(BrowserSwitchRequest, Intent)} upon return to the app.
+     * Returns null if an exception is thrown.
+     * @throws BrowserSwitchException when a browser is unable to be launched for browser switching
+     */
+    @Nullable
+    public BrowserSwitchRequest start(@NonNull ComponentActivity activity, @NonNull BrowserSwitchOptions browserSwitchOptions) throws BrowserSwitchException {
+        assertCanPerformBrowserSwitch(activity, browserSwitchOptions);
+
+        Uri browserSwitchUrl = browserSwitchOptions.getUrl();
+        int requestCode = browserSwitchOptions.getRequestCode();
+        String returnUrlScheme = browserSwitchOptions.getReturnUrlScheme();
+
+        JSONObject metadata = browserSwitchOptions.getMetadata();
+
+        BrowserSwitchRequest request;
+        if (activity.isFinishing()) {
+            String activityFinishingMessage =
+                    "Unable to start browser switch while host Activity is finishing.";
+            throw new BrowserSwitchException(activityFinishingMessage);
+        } else  {
+            boolean launchAsNewTask = browserSwitchOptions.isLaunchAsNewTask();
+            try {
+                 request =
+                        new BrowserSwitchRequest(requestCode, browserSwitchUrl, metadata, returnUrlScheme, true);
+                customTabsInternalClient.launchUrl(activity, browserSwitchUrl, launchAsNewTask);
+            } catch (ActivityNotFoundException e) {
+                throw new BrowserSwitchException("Unable to start browser switch without a web browser.");
+            }
+            return request;
+        }
+    }
+
+    void assertCanPerformBrowserSwitch(ComponentActivity activity, BrowserSwitchOptions browserSwitchOptions) throws BrowserSwitchException {
```

Comment:
random: I wonder if we should make `assertCanPerformBrowserSwitch()` public as an explicit callout to Kotlin merchants since the `start()` method without a try/catch will still compile in Kotlin.

Reply:
Hmmm... it feels like a bit of unnecessary lift for the merchant - should we instead of throwing an error in the `start` method return a `BrowserSwitchPendingRequest.Success` or `BrowserSwitchPendingRequest.Failure` object? Similar to how we will wrap it in the BT SDK

Reply:
Yeah that's a good call. Less API surface area too, so it's clear that it's the primary method for browser switching.

Reply:
Ok updated with the pending request type and removed throwing errors from this method all together. The next question will be how to parcelize/stringify the `BrowserSwitchPendingRequest` so that integrators can store it in whatever way they want. For our BT integration it doesn't matter but I've created a ticket to figure that out